pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -181,6 +181,16 @@ public void initialize() {
         }
     }
 
+    public Optional<LookupResult> getBrokerServiceUrl(TopicName topicName, LookupOptions options) {
+        try {
+            return getBrokerServiceUrlAsync(topicName, options).get();
```

Comment:
It's better not add this sync method anymore.

Reply:
Is it up to the caller to do the synchronous operation?

Reply:
change it to async